core/runtime/src/main/java/io/quarkus/runtime/configuration/ProfileManager.java

```diff
@@ -69,4 +76,14 @@ public static String getActiveProfile() {
         return launchMode.getDefaultProfile();
     }
 
+    public static List<String> getActiveProfiles() {
+        List<String> profiles = newCollectionConverter(newEmptyValueConverter(value -> value), ArrayList::new)
+                .convert(getActiveProfile());
```

Comment:
This will always return a list containing a single element? Why not simply `Collections.singletonList`?

Reply:
Also maybe it would be nice to cache this list for speed?

Reply:
I thought about caching it, but we were not caching the profile in `getActiveProfile`. Should we do it for both?

Reply:
I think so. This should save some nanoseconds ;)

Reply:
Having a further look into this, I'm not sure if it is worth it. The `setLaunchMode` affects the profile resolution, meaning that we cannot statically init the profiles.

Granted that the calls to `setLaunchMode` are somehow controlled, and we could possible lazy init the profiles without any issue. On the other hand we may want to use an AtomicReference to be on the safe side if we want to go down that road.

What do you think?

Reply:
That's fine